gum/backend-windows/gummemoryaccessmonitor-windows.c

```diff
@@ -242,14 +268,101 @@ gum_set_guard_flag (const GumLiveRangeDetails * details,
                     gpointer user_data)
 {
   GumMemoryAccessMonitor * self = GUM_MEMORY_ACCESS_MONITOR (user_data);
-  DWORD old_prot;
+  GumMemoryAccessMonitorPrivate * priv = self->priv;
+  DWORD old_prot, new_prot;
   BOOL success;
+  gboolean is_guarded = FALSE;
+  guint num_pages;
+  /* dummy init to make VS happy... */
+  new_prot = PAGE_NOACCESS;
+
+  if (priv->access_mask & GUM_PAGE_READ)
+  {
+    if (priv->auto_reset)
+    {
+      is_guarded = TRUE;
+      new_prot = details->prot | PAGE_GUARD;
+    }
+    else
+    {
+      new_prot = PAGE_NOACCESS;
+    }
+  }
+  else
+  {
+    switch (details->prot & 0xFF)
+    {
+    case PAGE_EXECUTE:
+      if (priv->access_mask & GUM_PAGE_EXECUTE)
+        new_prot = PAGE_READONLY;
+      else
+        return TRUE;
+      break;
+    case PAGE_EXECUTE_READ:
+      if (priv->access_mask & GUM_PAGE_EXECUTE)
+        new_prot = PAGE_READONLY;
+      else
+        return TRUE;
+      break;
+    case PAGE_EXECUTE_READWRITE:
+      if (priv->access_mask == GUM_PAGE_WRITE)
+        new_prot = PAGE_EXECUTE_READ;
+      else if (priv->access_mask == (GUM_PAGE_EXECUTE | GUM_PAGE_WRITE))
+        new_prot = PAGE_READONLY;
+      else if (priv->access_mask == GUM_PAGE_EXECUTE)
+        new_prot = PAGE_READWRITE;
+      else
+        g_assert_not_reached ();
+      break;
+    case PAGE_EXECUTE_WRITECOPY:
+      if (priv->access_mask == GUM_PAGE_WRITE)
+        new_prot = PAGE_EXECUTE_READ;
+      else if (priv->access_mask == (GUM_PAGE_EXECUTE | GUM_PAGE_WRITE))
+        new_prot = PAGE_READONLY;
+      else if (priv->access_mask == GUM_PAGE_EXECUTE)
+        new_prot = PAGE_WRITECOPY;
+      else
+        g_assert_not_reached ();
+      break;
+    case PAGE_NOACCESS:
+      return TRUE;
+    case PAGE_READONLY:
+      return TRUE;
+    case PAGE_READWRITE:
+      if (priv->access_mask & GUM_PAGE_WRITE)
+        new_prot = PAGE_READONLY;
+      else
+        return TRUE;
+      break;
+    case PAGE_WRITECOPY:
+      if (priv->access_mask & GUM_PAGE_WRITE)
+        new_prot = PAGE_READONLY;
+      else
+        return TRUE;
+      break;
+    default:
+      g_assert_not_reached ();
+    }
+  }
 
-  success = VirtualProtect (GSIZE_TO_POINTER (details->range->base_address),
-      details->range->size, details->prot | PAGE_GUARD, &old_prot);
+  num_pages = priv->num_pages;
+
+  priv->pages_details = g_realloc (priv->pages_details, 
+    (num_pages + 1)*sizeof(priv->pages_details[0]));
```

Comment:
- Missing spaces around `*`
- Missing space after sizeof
- Continuation (hanging indents) should be 4 spaces instead of 2